test/models/group_test.rb

```diff
@@ -206,4 +209,48 @@ def setup
   test '#metadata_summary incorporates fields from shared projects' do
     assert_equal %w[metadatafield1 metadatafield2], groups(:group_alpha).metadata_fields
   end
+
+  test 'group should have samples_count from all projects within' do
+    group12 = groups(:group_twelve)
+
+    project29 = projects(:project29)
+    Project.reset_counters(project29.id, :samples_count)
+    project29.reload.samples_count
+
+    project30 = projects(:project30)
+    Project.reset_counters(project30.id, :samples_count)
+    project30.reload.samples_count
+
+    project31 = projects(:project31)
+    Project.reset_counters(project31.id, :samples_count)
+    project31.reload.samples_count
+
+    expected_samples_count = group12.samples_count
+    actual_samples_count = Project.joins(:namespace).where(namespace: { parent_id: group12.self_and_descendants })
+                                  .select(:samples_count).pluck(:samples_count).sum
+
+    assert_equal expected_samples_count, actual_samples_count
+  end
```

Comment:
Didn't realize you had started this before writing the top comment, so yes change this to the above in my opinion.